frontend/src/components/JoinCode/JoinCode.js

```diff
@@ -40,11 +41,15 @@ const JoinCodeComponent = styled.div`
 `;
 
 function DisplayJoinCode({ loading, code }) {
+  const { state } = useContext(HostContext);
+  const hidden = state.gameSettings.hideJoinCode;
+
   return (
     <JoinCodeComponent>
       <p className="join-code-title">JOIN CODE:</p>
       <div className="join-code" data-testid="join-code">
-        {loading || !code ? <LoadingIndicator secondary /> : <p>{code}</p>}
+        {!hidden &&
+          (loading || !code ? <LoadingIndicator secondary /> : <p>{code}</p>)}
```

Comment:
I like ternaries but IMO this is really confusing. Creating a function that uses `if` statements might be more readable.

```
function joinCode() {
  if (!hidden && loading || !code) {
    return <LoadingIndicator secondary />
  }
  return <p>{code}</p>
}

{joinCode()}
```

Reply:
This *is* a common pattern you'll find in React code. I'm not sure it's worth abstracting simple inline logic to a function every time. I think we should let Cody comment on this.